ecosystem/typescript/sdk/src/aptos_client.ts

```diff
@@ -541,49 +573,48 @@ export class AptosClient {
-   * checking whether it was committed successfully. This has the same failure
-   * semantics as `submitTransaction` and `waitForTransactionWithResult`, see
-   * those for information about how this can fail (throw errors).
-   *
-   * If you set checkSuccess there are additional error cases, see the
-   * documentation for `checkSuccess` below.
+   * Helper for generating, signing, and submitting a transaction.
    *
-   * @param sender AptosAccount of transaction sender
-   * @param payload Transaction payload
-   * @param extraArgs Extra args for building transaction payload and configuring
-   * behavior of this function.
-   * @param checkSuccess If set, check whether the transaction was successful and
-   * throw a TransactionNotCommittedError if not.
+   * @param sender AptosAccount of transaction sender.
+   * @param payload Transaction payload.
+   * @param extraArgs Extra args for building the transaction payload.
    * @returns The transaction response from the API.
    */
-  async generateSignSendWaitForTransaction(
+  async generateSignSubmitTransaction(
     sender: AptosAccount,
     payload: TxnBuilderTypes.TransactionPayload,
     extraArgs?: {
       maxGasAmount?: BCS.Uint64;
       gasUnitPrice?: BCS.Uint64;
       expireTimestamp?: BCS.Uint64;
-      checkSuccess?: boolean;
     },
-  ): Promise<Gen.Transaction> {
-    /* eslint-disable max-len */
-    // :!:>generateSignSendWaitForTransactionInner
+  ): Promise<string> {
+    // :!:>generateSignSubmitTransactionInner
     const rawTransaction = await this.generateRawTransaction(sender.address(), payload, extraArgs);
     const bcsTxn = AptosClient.generateBCSTransaction(sender, rawTransaction);
     const pendingTransaction = await this.submitSignedBCSTransaction(bcsTxn);
-    const transactionResponse = await this.waitForTransactionWithResult(pendingTransaction.hash); // <:!:generateSignSendWaitForTransactionInner
-    /* eslint-enable max-len */
-    if (extraArgs?.checkSuccess === undefined || extraArgs?.checkSuccess === null || !extraArgs.checkSuccess) {
-      return transactionResponse;
-    }
-    if (!(transactionResponse as any)?.success) {
-      throw new TransactionNotCommittedError(
-        `Transaction ${pendingTransaction.hash} processed by blockchain but not committed successfully`,
-        transactionResponse,
-      );
-    }
-    return transactionResponse;
+    return pendingTransaction.hash; // <:!:generateSignSubmitTransactionInner
+  }
+
+  /**
+   * Helper for generating, submitting, and waiting for a transaction, and then
+   * checking whether it was committed successfully. Under the hood this is just
+   * `generateSignSubmitTransaction` and then `waitForTransactionWithResult`, see
+   * those for information about the return / error semantics of this function.
+   */
+  async generateSignSubmitWaitForTransaction(
+    sender: AptosAccount,
+    payload: TxnBuilderTypes.TransactionPayload,
```

Comment:
the extradArgs here doesn't match the one in generateSignSubmitTransaction.

Reply:
It's a combination of the extraArgs required for both functions. Since we only pull out the args necessary in those functions it's fine, there's no harm in passing in extra values (though it's a bit gross).